### PR TITLE
[FlagGems Operator Development Competition] Add softplus_backward and logit_backward

### PR DESCRIPTION
**Pull request by AdamPlatin123**
## Summary
- Add `softplus_backward` Triton kernel with ATen dispatch registration
- Add `logit_backward` Triton kernel with ATen dispatch registration

### softplus_backward
- Uses `pointwise_dynamic` pattern for element-wise computation
- Formula: `grad_input = grad_output * sigmoid(beta * x)` when `beta * x <= threshold`, else `grad_output`
- Supports configurable `beta` and `threshold` parameters

### logit_backward
- Formula: `grad_input = grad_output / (x * (1 - x))` when `x` is in `(eps, 1-eps)`, else `0`
- When `eps=None`: no clamping, directly computes `1 / (x * (1 - x))`
- When `eps` is specified: gradient is zero outside the valid range

## Testing
All tests passed on NVIDIA RTX PRO 6000 Blackwell with exact match against PyTorch ATen reference:
```
softplus_backward max diff: 0.000000e+00
logit_backward no eps max diff: 0.000000e+00
logit_backward eps=0.01 max diff: 0.000000e+00
```

Edge cases verified at boundary values (eps boundaries for logit, threshold for softplus).

## Hardware Environment
- GPU: NVIDIA RTX PRO 6000 Blackwell
- PyTorch: 2.11+cu128
- Triton: 3.6

### CODE REVIEW

**Comment by AdamPlatin123** [comment]
Ready for review — backward correctness verified on NVIDIA RTX PRO 6000 Blackwell. Thanks for your time!

**Comment by bin913** [comment]
@AdamPlatin123 Please resovle conflicts

**Comment by bin913** [comment]
@AdamPlatin123 Please add benchmark and test for new operator

**Comment by AdamPlatin123** [comment]
@bin913 Thank you for the review! I've resolved the merge conflicts by rebasing onto the latest master, and added benchmarks and tests for both `softplus_backward` and `logit_backward`:

- **Test**: `tests/test_softplus.py` — `test_softplus_backward`
- **Test**: `tests/test_logit.py` — `test_logit_backward`
- **Benchmark**: `benchmark/test_softplus.py` — `test_softplus_backward`
- **Benchmark**: `benchmark/test_logit.py` — `test_logit_backward`

All tests use autograd to validate against PyTorch reference implementations.
